leases/schema/mutations.py

```diff
@@ -41,37 +41,49 @@ class AbstractLeaseInput:
 
 class CreateBerthLeaseMutation(graphene.ClientIDMutation):
     class Input(AbstractLeaseInput):
-        application_id = graphene.ID(required=True)
+        # either application_id or customer_id must be provided, but not both
+        application_id = graphene.ID()
+        customer_id = graphene.ID()
         berth_id = graphene.ID(required=True)
 
     berth_lease = graphene.Field(BerthLeaseNode)
 
     @classmethod
-    @view_permission_required(BerthApplication, CustomerProfile)
-    @add_permission_required(BerthLease)
-    @transaction.atomic
-    def mutate_and_get_payload(cls, root, info, **input):
-        application = get_node_from_global_id(
-            info,
-            input.pop("application_id"),
-            only_type=BerthApplicationNode,
-            nullable=False,
-        )
+    def lookup_application_and_customer(cls, info, input):
+        from customers.schema import ProfileNode  # import here avoid circular import
 
-        if not application.customer:
-            raise VenepaikkaGraphQLError(
-                _("Application must be connected to an existing customer first")
+        if application_id := input.pop("application_id", None):
+            if "customer_id" in input:
+                raise VenepaikkaGraphQLError(
+                    _(
+                        "Can not specify both application and customer when creating a new berth lease"
+                    )
+                )
```

Comment:
This change might make it a bit clearer that both are not allowed simultaneously:

```suggestion
        application_id = input.pop("application_id", None)
        customer_id = input.pop("customer_id", None)
        
        if application_id and customer_id:
            raise VenepaikkaGraphQLError(
                _("Can not specify both application and customer when creating a new berth lease")
            )
        elif application_id:
            # ...
        elif customer_id:
            # ....
        else:
            # Raise
```